packages/golden-layout/src/items/Stack.ts

```diff
@@ -271,47 +323,52 @@ utils.copy(Stack.prototype, {
       contentItem.config[dimension] = 50;
       rowOrColumn.callDownwards('setSize');
     }
-  },
+  }
 
   /**
    * If the user hovers above the header part of the stack, indicate drop positions for tabs.
    * otherwise indicate which segment of the body the dragged item would be dropped on
    *
-   * @param    {Int} x Absolute Screen X
-   * @param    {Int} y Absolute Screen Y
-   *
-   * @returns {void}
+   * @param x Absolute Screen X
+   * @param y Absolute Screen Y
    */
-  _$highlightDropZone: function (x, y) {
-    var segment, area;
+  _$highlightDropZone(x: number, y: number) {
+    if (!this._contentAreaDimensions) {
+      return;
+    }
 
-    for (segment in this._contentAreaDimensions) {
-      area = this._contentAreaDimensions[segment].hoverArea;
+    for (let [segment, dimensions] of Object.entries(
+      this._contentAreaDimensions
+    )) {
+      const area = dimensions.hoverArea;
 
       if (area.x1 < x && area.x2 > x && area.y1 < y && area.y2 > y) {
         if (segment === 'header') {
           this._dropSegment = 'header';
-          this._highlightHeaderDropZone(x, y);
+          this._highlightHeaderDropZone(x);
         } else {
           this._resetHeaderDropZone();
-          this._highlightBodyDropZone(segment);
+          this._highlightBodyDropZone(segment as 'header'); // TS can't infer this is already 1 of the valid key strings
```

Comment:
I'll just do the type cast of `Object.entries`. It's somewhat annoying TS requires this even with types (not interfaces) since types aren't extendable/overridable without creating a new type. There's not any way to have TS infer the keys of the object that I could find